jsonpickle/handlers.py

```diff
@@ -215,8 +215,24 @@ def flatten(self, obj, data):
 
     def restore(self, data):
         restore = self.context.restore
-        factory, args = [restore(i, reset=False) for i in data['__reduce__']]
-        return factory(*args)
+        res = [restore(i, reset=False) for i in data['__reduce__']]
+        factory = res.pop(0)
+        args = res.pop(0)
+        obj = factory(*args)
+
+        if len(res) >= 1:
```

Comment:
I've merged this version as-is but please do submit another PR if you think we can improve this part even further.  Thanks for the careful review @marcintustin.